res/res_speech.c

```diff
@@ -201,7 +203,17 @@ struct ast_speech *ast_speech_new(const char *engine_name, const struct ast_form
 		if (ast_format_cap_iscompatible_format(engine->formats, ast_format_slin) != AST_FORMAT_CMP_NOT_EQUAL) {
 			best = ao2_bump(ast_format_slin);
 		} else {
-			return NULL;
+			/*
+			 * If there is no overlap and the engine does not support slin, find the best
+			 * format to translate to and set that as the 'best' input format for the engine.
+			 * speech_background will translate the channel via ast_set_read_format to this
```

Comment:
Don't reference the dialplan app, just state that the consumer of the API is responsible for translating to the chosen format.